src/stores/store.tsx

```diff
@@ -136,17 +138,37 @@ class Store {
         await this.disconnect(false);
       }
 
-      let ethersProvider: ethers.providers.JsonRpcProvider;
-      if (this.networkName === 'private') {
+      let ethersProvider:
+        | ethers.providers.JsonRpcProvider
+        | ethers.providers.Web3Provider;
```

Comment:
Web3Provider inherits from JsonRpcProvider, where do you need special access on web3Provider properties?